Fix neighborhood detail and wizard UX bugs

rachelekm opened the pull request:
## Overview


PR to fix a variety of small UX bugs, including:
- Allows 0 bedroom input when editing user profile
- Neighborhood detail: fixes transit map styling to display mbta map layers at lower zooms
- Neighborhood detail: fixes bug preventing transit map and trip routes from correctly updating when selecting neighborhood feature from the map
- Neighborhood detail: Fixes slightly-off map bounds in mobile trip route maps
- Neighborhood detail: Prevents "too far" route accordion panels from opening and displaying invalid route
- Neighborhood detail: Supports transit directions by ferry
- Fixes break line for trip length range indicator
- Prevents top 10 tour instructions from appearing over neighborhood details modal when refreshing page


### Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following
      [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [x] Run `./scripts/format` to lint, format, and fix the application source code.
- [x] CI passes after rebase

### Demo

https://github.com/user-attachments/assets/c4036cf8-9967-4ba4-8ae5-f912abfa7873


Fixed range line breaks:
<img width="304" height="175" alt="Screenshot 2025-10-28 at 1 11 38 PM" src="https://github.com/user-attachments/assets/ffbb8834-fc55-4b92-9c2c-aaa7157c7a4e" />
Enabled "ferry" transit option in trip directions:
<img width="376" height="669" alt="Screenshot 2025-10-28 at 1 12 05 PM" src="https://github.com/user-attachments/assets/eac39d0a-ec7e-4592-863a-ed031557bcae" />


## Testing Instructions

 * Login and confirm top 10 tour instructions display as expected on discover map view
 * Open edit profile wizard and confirm can continue past 1st step when inputting a 0 value for bedroom
 * Confirm can't input a non-number value or negative number to room input
 * Save wizard and confirm no regressions
 * Select neighborhood from map and confirm transit map at top of details displays all transit layers
 * Close modal and select different neighborhood from map
 * Confirm all parts of neighborhood detail modal update, specifically: transit map bounds are fit to correct neighborhood bounds in top map and all trip routes have correct start/end points for new neighborhood selection
 * Open each trip accordion and confirm: 1. maps are not loaded until accordion is opened, 2. maps correctly fit bounds to each accordion item route
 * On desktop view, confirm no regressions displaying different routes on one map as different accordions are expanded
 * Confirm that longer commute time ranges display ellipses vs wrapping around in your trips section as well as in neighborhood preview cards
 * Back on mobile, refresh page while opened neighborhood detail modal and confirm top 10 tour instructions do not open on top of details modal
 * Close modal and confirm top 10 instructions display on map view
 * Add 700 boylston st destination and then select "Hull" neighborhood
 * Confirm this trip destination includes a ferry route leg and "HINGHAM" correctly displays in the map trip directions


Resolves #747 #748 #761 #759

rachelekm committed:
Issue 747: Fix disabled 0 room input

rachelekm committed:
Issue 748: Display mbta layers at lower zoom in detail

rachelekm committed:
Issue 748: Ensure detail refreshes on open from map

rachelekm committed:
Issue 759: Fix trip map bounds on accordion open

rachelekm committed:
Fix too far route displaying in details

rachelekm committed:
Support ferry transit route path

rachelekm committed:
Issue 761: Fix wrapping range labels

rachelekm committed:
Prevent top 10 tour instructions on detail start

aaronxsu commented on src/frontend/src/components/base/Accordion/AccordionItem.tsx:
```diff
@@ -26,12 +27,20 @@ export const AccordionItem = ({
+            id={props.id}
+            className={styles.accordionItemWrapper()}
+            onExpandedChange={isExpanded =>
+                lazyLoad && isExpanded && !isLazyLoaded && setIsLazyLoaded(true)
```
It took me a bit to figure out why we have both `lazyLoad` and `isLazyLoaded`, and also what they mean here in this line. It might be helpful to add some quick comments about them.

rachelekm replied:
Added a clarifying comment in [41d58df](https://github.com/azavea/echo-locator/pull/762/commits/41d58df702770d5bf7ead7be807dfccad3087013)

aaronxsu commented on src/frontend/src/components/YourTrips/TripMap.tsx:
```diff
@@ -95,13 +100,18 @@ const TripMap = ({
                 interactive={false}
                 // @ts-ignore
-                mapStyle={baseMapDetailStyle}
+                mapStyle={baseMapTripStyle}
```
It might be helpful to add a comment here how baseMapTripStyle differs from baseMapDetailStyle _or_ baseMapStyle, since it's such a dense file.

rachelekm replied:
Added clarifying comment in [5aa0943](https://github.com/azavea/echo-locator/pull/762/commits/5aa0943b7f2b52255bd6b3a7deaff765a1f8af28)

aaronxsu reviewed (approved):
Works smoothly as advertised. I left two small suggestions to add code comments.

Another thing I found is that the style of the safety range label can sometimes look like the following. If this can be reproduced on your end, it might worth taking a look

<img width="328" height="685" alt="Image" src="https://github.com/user-attachments/assets/7fd17d16-d8ff-424d-a3d4-8c6ff8a51f9e" />

Other than that, this is good to go.

rachelekm committed:
Update CHANGELOG

rachelekm commented:
Thanks, @aaronxsu! I was able to reproduce the ranges falling out of alignment and added a quick fix as part of [d625183](https://github.com/azavea/echo-locator/pull/762/commits/d625183ce3600e42c8f04c250b7e13ccfe0dbfdb).  I also added clarifying comments to the two snippets that needed more context. I added a changelog entry and will merge in on succesful CI
